geochem_inverse_optimize.py

```diff
@@ -103,14 +108,22 @@ def get_sample_graphs(data_dir: str) -> Tuple[nx.DiGraph, "pyfastunmix.SampleAdj
 
 
 class SampleNetwork:
+    # TODO Docstrings
     def __init__(
         self,
         sample_network: nx.DiGraph,
-        sample_adjacency: "pyfastunmix.SampleAdjacency",
+        sample_adjacency: "pyfastunmix.SampleAdjacency" = None,
```

Comment:
If you're setting this to `None` then the type needs to be `Optional["pyfastunmix.SampleAdjacency"]`.